storage/grpc_client.go

```diff
@@ -156,8 +157,52 @@ func (c *grpcStorageClient) CreateBucket(ctx context.Context, project string, at
 	return battrs, err
 }
 
-func (c *grpcStorageClient) ListBuckets(ctx context.Context, project string, opts ...storageOption) (*BucketIterator, error) {
-	return nil, errMethodNotSupported
+func (c *grpcStorageClient) ListBuckets(ctx context.Context, project string, opts ...storageOption) *BucketIterator {
+	s := callSettings(c.settings, opts...)
+	it := &BucketIterator{
```

Comment:
Wdym by that? I think we need to create it here in order initialize the BucketIterator's PageInfo and nextFunc, and we also need to be able to capture it in the function closure created when defining `fetch`.

Reply:
Ah, I meant that since this code is duplicated we could just create it in the top level method and then pass `it *BucketIterator` as a param to the interface method.

Reply:
Ah ok I misunderstood. I think I'd like to keep this code in the transport-specific implementations just to keep the interface a little cleaner. It feels like a proper separation of concerns with this code here. Happy to change it though if you'd prefer.

Reply:
That sounds reasonable!